Documentation/app-operations.md

```diff
@@ -0,0 +1,234 @@
+# Fine-Grained Pod Manipulations
+
+To provide imperative application capabilities (e.g. start|stop|add) inside a
+pod for finer-grained control over rkt containerization concepts and debugging
+needs, this proposal introduces new stage1 entrypoints that will be used for
+manipulating applications inside pods.
+
+The motivation behind this change is the direction orchestration systems are
+taking and to facilitate this new direction. For more details, see
+[kubernetes#25899](https://github.com/yujuhong/kubernetes/blob/08dc66113399c89e31f6872f3c638695a6ec6a8d/docs/proposals/container-runtime-interface-v1.md).
+
+# API
+
+The envisioned workflow for the app-level CLI API is that After a pod has been started, 
+the users will be invoking the rkt CLI which is application logic on top of stage1 entrypoints (introduced below).
+
+## rkt app add
+Injects an application image into a running pod.
+
+It first prepares an application rootfs in stage0 for the application image/images
+and then the prepared-app is injected via the entrypoint.
+
+```bash
+rkt app add <pod-uuid> <image-name/hash/address/registry-URL> <arguments>
```

Comment:
I think the former choice makes operations easier and a bit more convenient, especially for collisions